### DISCUSSION TITLE
[General] When will the forking function of hardhat be supported?

### DISCUSSION THREAD

**Discussion by EraHermes** [zkSync-Community-Hub/zksync-developers#29] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Does Zksync Era support hardhat forking, I need it for code testing

### What is the specific issue or error you're encountering?

I'm currently working on some arbitrage programs, and I can't verify if my programs work

### Can you share the error messages or logs you're receiving, if any?

Nope

### Have you made any recent changes to the contract before encountering this issue?

Nope

### Are there any external libraries or contracts that your contract interacts with?

Nope

### Can you provide the relevant portions of your contract code where the issue is occurring?

Nope

### Have you tried to isolate the problem, and if so, what were the results?

Nope

### What steps have you already taken to try to resolve the issue?

Nope

**Comment by dutterbutter** [comment, marked as answer]
We currently do not support hardhat forking but we are actively working on it. We understand its importance and will provide an update once support lands.

**Reply by Vorh** [reply]
@dutterbutter Hello! May I know the estimated timeline for this feature?

**Reply by uF4No** [reply]
You can fork other networks using era-test-node in Hardhat. Find the details in [this section of the docs](https://docs.zksync.io/build/test-and-debug/era-test-node.html#forking-networks)

**Comment by steyn52** [comment]
As of my last update, zkSync Era does not natively support forking in Hardhat for testing purposes. However, you can simulate testing scenarios by using the available tools and environments provided by zkSync, such as their testnets or local development environments. Keep an eye on any updates or announcements from zkSync for potential future support for forking in Hardhat.

**Comment by cameronstroman** [comment]
Functionality: Testing arbitrage programs with Hardhat forking on Zksync Era Mainnet.
Issue: Unable to determine if Zksync Era supports Hardhat forking for testing.

**Comment by unticcutecerin** [comment]
The user is asking if zkSync Era supports hardhat forking for code testing purposes. They are working on an arbitrage program but cannot verify if it works. The user has not provided any error messages, logs, or code snippets, making it difficult to provide a specific solution. However, it may be helpful for the user to consult the zkSync documentation or reach out to the zkSync support team for guidance on how to use hardhat forking with zkSync Era.
